src/main/java/com/webapp/TextBook/repository/data_access/BookCopy.java

```diff
@@ -0,0 +1,551 @@
+
+    /**
+     *<p>
+     *     Sets the editionYear to a new date.
+     *</p>
+     *
+     * @param editionYear: int editionYear variable
+     */
+    public void setEditionYear(int editionYear) {
+        this.editionYear = editionYear;
+    }
+
+    /**
+     *<p>
+     *     Sets the pidm to a new string.
+     *</p>
+     *
+     * @param pidm: String pidm variable
+     */
+    public void setPidm(String pidm) {
+        this.pidm = pidm;
+    }
+
+    /**
+     *<p>
+     *     Sets the prevDateCheckedIn to a new date.
+     *</p>
+     *
+     * @param prevDateCheckedIn: Date prevDateCheckedIn variable
+     */
+    public void setPrevDateCheckedIn(Date prevDateCheckedIn) {
+        this.prevDateCheckedIn = prevDateCheckedIn;
+    }
+
+    /**
+     *<p>
+     *     Sets the prevPidm to a new double.
+     *</p>
+     *
+     * @param prevPidm: double prevPidm variable
+     */
+    public void setPrevPidm(double prevPidm) {
+        this.prevPidm = prevPidm;
+    }
+
+    /**
+     *<p>
+     *     Sets the prevTermCode to a new string.
+     *</p>
+     *
+     * @param prevTermCode: String prevTermCode variable
+     */
+    public void setPrevTermCode(String prevTermCode) {
+        this.prevTermCode = prevTermCode;
+    }
+
+    /**
+     *<p>
+     *     Sets the seqNr to a new int.
+     *</p>
+     *
+     * @param seqNr: int seqNr variable
+     */
+    public void setSeqNr(int seqNr) {
+        this.seqNr = seqNr;
+    }
+
+    /**
+     *<p>
+     *     Sets the strikeBarcode to a new string.
+     *</p>
+     *
+     * @param strikeBarcode: String strikeBarcode variable
+     */
+    public void setStrikeBarcode(String strikeBarcode) {
+        this.strikeBarcode = strikeBarcode;
+    }
+
+    /**
+     *<p>
+     *     Sets the termCode to a new string.
+     *</p>
+     *
+     * @param termCode: String termCode variable
+     */
+    public void setTermCode(String termCode) {
+        this.termCode = termCode;
+    }
+
+    /**
+     *<p>
+     *     Sets the _ID to a new UUID.
+     *</p>
+     *
+     * @param _ID: UUID _ID variable
+     */
+    public void set_ID(UUID _ID) {
+        this._ID = _ID;
+    }
+
+    /**
+     *<p>
+     * Updates a Java BookCopy object with the values given from the DBO object information.
+     *</p>
+     *
+     * @param values: Generic object holding the DBO information of a BookCopy
+     */
+    @Override
+    public void updateDataAccessObject(@NotNull Object[] values) {
+        // order: pkey (bookCode, editionYear, seqNr), strikeBarcode, dateCheckedOut, disposition,
+        //bookSalePrice, prevPidm, prevTermCode, prevDateCheckedIn, activityDate, billFlag
+        this.bookCode = (String)values[0];
+        this.editionYear = ((BigDecimal)values[1]).intValue();
+        this.seqNr = ((BigDecimal)values[2]).intValue();
+        this.strikeBarcode = (String)values[3];
+        this.dateCheckedOut =(Date)values[4];
+        this.disposition = ((String)values[5]).charAt(0);
```

Comment:
why are we getting char at 0 -- what does the dbo provide?

Reply:
Added comment line indicating the correspondence between the values[] index integers and table columns.

Reply:
https://github.com/NicholasPierce1/TextBookServices/pull/33/commits/49f81cdcc44c2010d65bfc029af433c25858840e